src/abstractdataframe/abstractdataframe.jl

```diff
@@ -368,34 +368,42 @@ function Base.dump(io::IO, df::AbstractDataFrame, n::Int, indent)
     end
 end
 
-# summarize the columns of a df
-# TODO: clever layout in rows
+
 """
-Summarize the columns of an AbstractDataFrame
+Report descriptive statistics for a DataFrame
 
 ```julia
-describe(df::AbstractDataFrame)
-describe(io, df::AbstractDataFrame)
+describe(df::AbstractDataFrame; stats = [:mean, :min, :median, :max, :nmissing, :datatype])
+describe(io, df::AbstractDataFrame; stats = [:mean, :min, :median, :max, :nmissing, :datatype])
 ```
 
 **Arguments**
 
 * `df` : the AbstractDataFrame
 * `io` : optional output descriptor
+* `stats::AbstractVector{Symbol}`: the summary statistics to report. Allowed 
+  fields are `:mean, :std, :min, :q25, :median, :q75, :max, :eltype, :nunique, 
+  and :nmissing`
 
 **Result**
 
-* nothing
+* A `DataFrame` where each row represents a variable and each column a summary statistic.
 
 **Details**
 
-If the column's base type derives from Number, compute the minimum, first
-quantile, median, mean, third quantile, and maximum. Missings are filtered and
-reported separately.
+If the column's base type derives from `Number`, compute the mean, standard 
+deviation, minimum, first quantile, median, third quantile, and maximum. If 
+a column is not numeric, these statistics are populated with `nothing`s. 
 
-For boolean columns, report trues, falses, and missings.
+For non-numeric types, `describe` can also report the number of missing values 
+with `nunique` specified in the `stats` keyword argument. For numeric types, 
+`nunique` will return `nothing`s.
 
-For other types, show column characteristics and number of missings.
+Missing values are filtered in the calculation of all statistics, however the column
+`nmissing` will report the number of missing values of that variable. 
+If the column does not allow missing values, `nothing` is returned. 
+Consequently, `nmissing = 0` indicates that the column allows
+missing values, but does not contain any at the time. 
```

Comment:
I'm not a native English speaker, but "at the time" sounds weird. Maybe "currently"?

Reply:
It is definitely bad english. Changed to currently.